File elide-datastore-aggregation Javadoc build warnings

Fix the following javadoc build warnings for the aggregation data store:

```
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java:103: @return tag should be present and have description. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java:114: @return tag should be present and have description. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java:124: @return tag should be present and have description. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java:156: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryValidator.java:140: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/ColumnMeta.java:32: @return tag should be present and have description. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/TableMeta.java:43: @return tag should be present and have description. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/JoinPath.java:41: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/QueryLogger.java:17: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/QueryLogger.java:22: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/QueryLogger.java:34: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/QueryLogger.java:50: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/QueryResponse.java:11: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/Slf4jQueryLogger.java:22: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/MetaDataStore.java:202: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/MetaDataStore.java:236: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/MetaDataStore.java:316: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/models/Column.java:123: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/models/Column.java:135: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/models/Metric.java:69: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/ColumnProjection.java:102:19: Expected @throws tag for 'UnsupportedOperationException'. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/ColumnProjection.java:112:19: Expected @throws tag for 'UnsupportedOperationException'. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/ColumnProjection.java:115: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/Queryable.java:29: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/Queryable.java:38: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/Queryable.java:46: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/NamedParamPreparedStatement.java:25: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/annotation/FromSubquery.java:31: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/annotation/FromTable.java:31: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/annotation/VersionQuery.java:27: @return tag should be present and have description. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/dialects/SQLDialect.java:22: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/DynamicSQLReferenceTable.java:29: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceTable.java:76: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceVisitor.java:36: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceVisitor.java:52: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceVisitor.java:84: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceVisitor.java:154: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/SQLColumnProjection.java:22: @return tag should be present and have description. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/SQLMetricProjection.java:47:12: More than 7 parameters (found 8). [ParameterNumber]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/timegrains/Time.java:127:12: More than 7 parameters (found 8). [ParameterNumber]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/timegrains/Time.java:145:12: More than 7 parameters (found 8). [ParameterNumber]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java:59:44: Expected @throws tag for 'Exception'. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java:90: First sentence should end with a period. [JavadocStyle]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java:94:43: Expected @throws tag for 'Exception'. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java:218:39: Expected @throws tag for 'Exception'. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java:257:41: Expected @throws tag for 'Exception'. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java:341:52: Expected @throws tag for 'Exception'. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java:381:44: Expected @throws tag for 'Exception'. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java:436:41: Expected @throws tag for 'Exception'. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java:476:48: Expected @throws tag for 'Exception'. [JavadocMethod]
[WARN] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java:509:56: Expected @throws tag for 'Exception'. [JavadocMethod]
```
and

```
[WARNING] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java:101: warning: no description for @return
[WARNING] * @return
[WARNING] ^
[WARNING] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java:112: warning: no description for @return
[WARNING] * @return
[WARNING] ^
[WARNING] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java:122: warning: no description for @return
[WARNING] * @return
[WARNING] ^
[WARNING] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java:169: warning: no description for @param
[WARNING] * @param transaction
[WARNING] ^
[WARNING] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/QueryVisitor.java:11: warning: no description for @param
[WARNING] * @param <T>
[WARNING] ^
[WARNING] /sd/workspace/src/github.com/yahoo/elide/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/SQLColumnProjection.java:20: warning: no description for @return
```

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryEngine.java
@@ -25,6 +25,7 @@ import lombok.Getter;
 
 import java.util.List;
 import java.util.Map;
+
 /**
  * A {@link QueryEngine} is an abstraction that an AggregationDataStore leverages to run analytic queries (OLAP style)
  * against an underlying persistence layer.
@@ -98,7 +99,7 @@ public abstract class QueryEngine {
      * @param dimension The dimension column.
      * @param alias The client provide alias.
      * @param arguments The client provided parameterized arguments.
-     * @return
+     * @return DimensionProjection
      */
     public abstract ColumnProjection constructDimensionProjection(Dimension dimension,
                                                                   String alias,
@@ -109,7 +110,7 @@ public abstract class QueryEngine {
      * @param dimension The dimension column.
      * @param alias The client provide alias.
      * @param arguments The client provided parameterized arguments.
-     * @return
+     * @return TimeDimensionProjection
      */
     public abstract TimeDimensionProjection constructTimeDimensionProjection(TimeDimension dimension,
                                                                              String alias,
@@ -119,7 +120,7 @@ public abstract class QueryEngine {
      * @param metric The metric column.
      * @param alias The client provide alias.
      * @param arguments The client provided parameterized arguments.
-     * @return
+     * @return MetricProjection
      */
     public abstract MetricProjection constructMetricProjection(Metric metric,
                                                                String alias,
@@ -154,7 +155,7 @@ public abstract class QueryEngine {
         void close();
 
         /**
-         * Cancels running transaction
+         * Cancels running transaction.
          */
         void cancel();
     }

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryValidator.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryValidator.java
@@ -131,7 +131,7 @@ public class QueryValidator {
     }
 
     /**
-     * Verifies that the current path can be sorted on
+     * Verifies that the current path can be sorted on.
      * @param path The path that we are validating
      * @param allFields Set of all field names included in initial query
      */

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/ColumnMeta.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/ColumnMeta.java
@@ -28,6 +28,7 @@ public @interface ColumnMeta {
 
     /**
      * Indicates the cardinality for the column.
+     * @return size
      */
     CardinalitySize size() default CardinalitySize.UNKNOWN;
 }

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/DimensionFormula.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/DimensionFormula.java
@@ -72,6 +72,7 @@ public @interface DimensionFormula {
 
     /**
      * The arguments accepted by this table.
+     * @return arguments for the dimension
      */
     ArgumentDefinition[] arguments() default {};
 }

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/MetricFormula.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/MetricFormula.java
@@ -73,6 +73,7 @@ public @interface MetricFormula {
 
     /**
      * The arguments accepted by this table.
+     * @return arguments for the metric
      */
     ArgumentDefinition[] arguments() default {};
 }

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/TableMeta.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/annotation/TableMeta.java
@@ -39,11 +39,13 @@ public @interface TableMeta {
 
     /**
      * Indicates the size of the table.
+     * @return size
      */
     CardinalitySize size() default CardinalitySize.UNKNOWN;
 
     /**
      * The arguments accepted by this table.
+     * @return arguments for the table
      */
     ArgumentDefinition[] arguments() default {};
 }

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/JoinPath.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/JoinPath.java
@@ -41,7 +41,7 @@ public class JoinPath extends Path {
     }
 
     /**
-     * Extend this path with a extension dot separated path
+     * Extend this path with a extension dot separated path.
      *
      * @param extensionPath extension path append to this join path
      * @param dictionary dictionary

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/QueryLogger.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/QueryLogger.java
@@ -15,12 +15,12 @@ import java.util.UUID;
 import javax.ws.rs.core.MultivaluedMap;
 
 /**
- * Query Logger Interface for Elide
+ * Query Logger Interface for Elide.
  */
 public interface QueryLogger {
 
     /**
-     * Accepts the incoming JSON API query and notes the start time for the query
+     * Accepts the incoming JSON API query and notes the start time for the query.
      * @param queryId The RequestScope requestId.
      * @param user The Principal user
      * @param headers Http Request Headers
@@ -32,7 +32,7 @@ public interface QueryLogger {
                      Optional<MultivaluedMap<String, String>> queryParams, String path);
 
     /**
-     * Processes and logs all the queries from QueryDetail
+     * Processes and logs all the queries from QueryDetail.
      * @param queryId The RequestScope requestId.
      * @param query The underlying Query
      * @param apiQuery The output querytext
@@ -41,14 +41,14 @@ public interface QueryLogger {
     void processQuery(UUID queryId, Query query, List<String> apiQuery, boolean isCached);
 
     /**
-     * Cancels all queries currently running for a particular requestId
+     * Cancels all queries currently running for a particular requestId.
      * Implementation must be thread-safe.
      * @param queryId The RequestScope requestId.
      */
     void cancelQuery(UUID queryId);
 
     /**
-     * Registers the endtime for a query and logs it out
+     * Registers the endtime for a query and logs it out.
      * @param queryId The RequestScope requestId.
      * @param response The ElideResponse object
      */

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/QueryResponse.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/QueryResponse.java
@@ -9,7 +9,7 @@ import lombok.AllArgsConstructor;
 import lombok.Getter;
 
 /**
- * QueryLogger Response Class
+ * QueryLogger Response Class.
  */
 @AllArgsConstructor
 public class QueryResponse {

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/Slf4jQueryLogger.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/core/Slf4jQueryLogger.java
@@ -20,7 +20,7 @@ import java.util.UUID;
 import javax.ws.rs.core.MultivaluedMap;
 
 /**
- * Default Slf4j Logging Implementation of QueryLogger for Elide
+ * Default Slf4j Logging Implementation of QueryLogger for Elide.
  */
 @Slf4j
 public class Slf4jQueryLogger implements QueryLogger {

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/MetaDataStore.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/MetaDataStore.java
@@ -189,7 +189,7 @@ public class MetaDataStore implements DataStore {
     }
 
     /**
-     * Get a table metadata object
+     * Get a table metadata object.
      *
      * @param tableClass table class
      * @return meta data table
@@ -223,7 +223,7 @@ public class MetaDataStore implements DataStore {
     }
 
     /**
-     * Get a {@link Column} for the last field in a {@link Path}
+     * Get a {@link Column} for the last field in a {@link Path}.
      *
      * @param path path to a field
      * @return meta data column
@@ -293,7 +293,7 @@ public class MetaDataStore implements DataStore {
     }
 
     /**
-     * Get all metadata of a specific metadata class
+     * Get all metadata of a specific metadata class.
      *
      * @param cls metadata class
      * @param <T> metadata class

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/models/Column.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/models/Column.java
@@ -143,7 +143,7 @@ public abstract class Column implements Versioned {
     }
 
     /**
-     * Construct a column name as meta data
+     * Construct a column name as meta data.
      *
      * @param tableClass table class
      * @param fieldName field name
@@ -155,7 +155,7 @@ public abstract class Column implements Versioned {
     }
 
     /**
-     * Resolve the value type of a field
+     * Resolve the value type of a field.
      *
      * @param tableClass table class
      * @param fieldName field name

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/ColumnProjection.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/ColumnProjection.java
@@ -83,7 +83,7 @@ public interface ColumnProjection extends Serializable {
     int hashCode();
 
     /**
-     * Creates an alias that is not vulnerable to injection
+     * Creates an alias that is not vulnerable to injection.
      * @param name projected column's name
      * @param alias projected column's alias
      * @return an alias for projected column that is not vulnerable to injection

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/Optimizer.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/Optimizer.java
@@ -19,7 +19,7 @@ public interface Optimizer {
     boolean canOptimize(Query query, SQLReferenceTable referenceTable);
 
     /**
-     * Optimizes the query
+     * Optimizes the query.
      * @param query The query to optimize.
      * @return A new optimized query.
      */

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/Queryable.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/query/Queryable.java
@@ -29,7 +29,7 @@ public interface Queryable {
     Queryable getSource();
 
     /**
-     * Every queryable needs an alias which uniquely identifies the queryable in an individual query
+     * Every queryable needs an alias which uniquely identifies the queryable in an individual query.
      * @return The alias
      */
     default String getAlias() {
@@ -38,7 +38,7 @@ public interface Queryable {
     }
 
     /**
-     * The name of the queryable
+     * The name of the queryable.
      * @return The name
      */
     default String getName() {
@@ -46,7 +46,7 @@ public interface Queryable {
     }
 
     /**
-     * The version of the queryable
+     * The version of the queryable.
      * @return The version
      */
     default String getVersion() {

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/NamedParamPreparedStatement.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/NamedParamPreparedStatement.java
@@ -23,7 +23,7 @@ import java.util.regex.Pattern;
 public class NamedParamPreparedStatement {
 
     /**
-     * Parameter Pattern as defined in {@link FilterPredicate#getParameters()}
+     * Parameter Pattern as defined in {@link FilterPredicate#getParameters()}.
      * examples:
      * :overallRating_c82e10a5_0
      * :lowScore_7c4e440_0

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/annotation/FromSubquery.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/annotation/FromSubquery.java
@@ -29,7 +29,7 @@ public @interface FromSubquery {
     String sql();
 
     /**
-     * DB Connection Name for this query
+     * DB Connection Name for this query.
      * @return String DB Connection Name
      */
     // TO DO

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/annotation/FromTable.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/annotation/FromTable.java
@@ -29,7 +29,7 @@ public @interface FromTable {
     String name();
 
     /**
-     * DB Connection Name for this table
+     * DB Connection Name for this table.
      * @return String DB Connection Name
      */
     // TO DO

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/annotation/VersionQuery.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/annotation/VersionQuery.java
@@ -23,6 +23,7 @@ public @interface VersionQuery {
 
     /**
      * A String containing an SQL SELECT statement.
+     * @return SQL query String
      */
     String sql();
 }

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/dialects/SQLDialect.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/dialects/SQLDialect.java
@@ -20,7 +20,7 @@ public interface SQLDialect {
     String getDialectType();
 
     /**
-     * Checks whether we need to use alias for orderby
+     * Checks whether we need to use alias for orderby.
      * @return boolean.
      */
     boolean useAliasForOrderByClause();

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/DynamicSQLReferenceTable.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/DynamicSQLReferenceTable.java
@@ -28,7 +28,7 @@ public class DynamicSQLReferenceTable extends SQLReferenceTable {
     }
 
     /**
-     * Get the resolved physical SQL reference for a field from storage
+     * Get the resolved physical SQL reference for a field from storage.
      *
      * @param queryable table class
      * @param fieldName field name

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceTable.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceTable.java
@@ -80,7 +80,7 @@ public class SQLReferenceTable {
     }
 
     /**
-     * Get the resolved physical SQL reference for a field from storage
+     * Get the resolved physical SQL reference for a field from storage.
      *
      * @param queryable table class
      * @param fieldName field name

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceVisitor.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/metadata/SQLReferenceVisitor.java
@@ -40,7 +40,7 @@ public class SQLReferenceVisitor extends ColumnVisitor<String> {
     }
 
     /**
-     * For physical reference, just append it to the table alias
+     * For physical reference, just append it to the table alias.
      *
      * @param reference physical column name
      * @return <code>table.reference</code>
@@ -59,7 +59,7 @@ public class SQLReferenceVisitor extends ColumnVisitor<String> {
     }
 
     /**
-     * For a FIELD dimension, append its physical columnName to the table alias
+     * For a FIELD dimension, append its physical columnName to the table alias.
      *
      * @param source The parent which owns the dimension.
      * @param dimension a FIELD dimension
@@ -96,7 +96,7 @@ public class SQLReferenceVisitor extends ColumnVisitor<String> {
     }
 
     /**
-     * For a FORMULA column, resolve each reference individually and
+     * For a FORMULA column, resolve each reference individually and.
      *
      * @param source The parent which owns the column.
      * @param column
@@ -161,7 +161,7 @@ public class SQLReferenceVisitor extends ColumnVisitor<String> {
     }
 
     /**
-     * Get name for the last field in a {@link Path}
+     * Get name for the last field in a {@link Path}.
      *
      * @param path path to a field
      * @return field name

--- a/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/SQLColumnProjection.java
+++ b/elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/query/SQLColumnProjection.java
@@ -25,7 +25,7 @@ public interface SQLColumnProjection extends ColumnProjection {
      * Generate a SQL fragment for this combination column and client arguments.
      * @param source the queryable that contains the column.
      * @param table symbol table to resolve column name references.
-     * @return
+     * @return SQL query String for this column
      */
     default String toSQL(Queryable source, SQLReferenceTable table) {
         return table.getResolvedReference(source, getName());

--- a/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java
+++ b/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/QueryEngineTest.java
@@ -54,6 +54,8 @@ public class QueryEngineTest extends SQLUnitTest {
 
     /**
      * Test loading all three records from the table.
+     *
+     * @throws Exception exception
      */
     @Test
     public void testFullTableLoad() throws Exception {
@@ -88,7 +90,9 @@ public class QueryEngineTest extends SQLUnitTest {
     }
 
     /**
-     * Test loading records using {@link FromSubquery}
+     * Test loading records using {@link FromSubquery}.
+     *
+     * @throws Exception exception
      */
     @Test
     public void testFromSubQuery() throws Exception {
@@ -213,6 +217,8 @@ public class QueryEngineTest extends SQLUnitTest {
 
     /**
      * Test sorting by dimension attribute which is not present in the query.
+     *
+     * @throws Exception exception
      */
     @Test
     public void testSortJoin() throws Exception {
@@ -252,6 +258,8 @@ public class QueryEngineTest extends SQLUnitTest {
 
     /**
      * Test pagination.
+     *
+     * @throws Exception exception
      */
     @Test
     public void testPagination() throws Exception {
@@ -336,6 +344,8 @@ public class QueryEngineTest extends SQLUnitTest {
 
     /**
      * Test sorting by two different columns-one metric and one dimension.
+     *
+     * @throws Exception exception
      */
     @Test
     public void testSortByMultipleColumns() throws Exception {
@@ -376,6 +386,8 @@ public class QueryEngineTest extends SQLUnitTest {
 
     /**
      * Test grouping by a dimension with a JoinTo annotation.
+     *
+     * @throws Exception exception
      */
     @Test
     public void testJoinToGroupBy() throws Exception {
@@ -431,6 +443,8 @@ public class QueryEngineTest extends SQLUnitTest {
 
     /**
      * Test grouping by a dimension with a JoinTo annotation.
+     *
+     * @throws Exception exception
      */
     @Test
     public void testJoinToSort() throws Exception {
@@ -471,6 +485,8 @@ public class QueryEngineTest extends SQLUnitTest {
 
     /**
      * Test month grain query.
+     *
+     * @throws Exception exception
      */
     @Test
     public void testTotalScoreByMonth() throws Exception {
@@ -504,6 +520,8 @@ public class QueryEngineTest extends SQLUnitTest {
 
     /**
      * Test filter by time dimension.
+     *
+     * @throws Exception exception
      */
     @Test
     public void testFilterByTemporalDimension() throws Exception {

